### PR TITLE
Recursive Solution of Kth_Symbol_Grammar Added

### PR DESCRIPTION
**Pull request by Ayush12062000**
Hi, I am Ayush Kesarwani, participating in GSSOC'21 .
I'm adding C++ code (Recursive solution) for finding **K-th symbol in grammar** to this project.

Kindly review and merge this PR!

### CODE REVIEW

**Comment by tarun26091999** [comment]
mention the issue.

**Comment by Ayush12062000** [comment]
Find K-th Symbol in Grammar. #645


https://github.com/Algo-Phantoms/Algo-Tree/issues/645

File: Code/C++/Kth_Symbol_Grammar.cpp [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+        // Printing returned value from the function (either 0 or 1)
+
+    }
+	return 0;
+}
+
+
+/*
+
+	Test cases: 4
```
Comment (krishna-NIT) [inline Code/C++/Kth_Symbol_Grammar.cpp:62]: Add Space and time Complexity

File: Code/C++/Kth_Symbol_Grammar.cpp [hunk 2]
```diff
@@ -0,0 +1,72 @@
+/*
+	Problem Statement:- First row has 0. Every 0 is changed to 01 and every 1 is changed to 10.
+	Given n rows and index K (column), return the K-th indexed symbol in row n.
+
```
Comment (krishna-NIT) [inline Code/C++/Kth_Symbol_Grammar.cpp:4]: Explain in Brief your Logic/ Approach

**Comment by krishna-NIT** [comment]
Add the issue name and link the location of code in proper readme, for this refer any previous merged PR,
How and which readme is to be updated

**Comment by Ayush12062000** [comment]
okay, sir will make the necessary changes ASAP.

File: Algorithm/Recursion/readme.md [hunk 3]
```diff
@@ -18,3 +18,4 @@ eventually converge on the base case.
 * Factorial ----> [C++](https://github.com/Algo-Phantoms/Algo-Tree/blob/main/Code/C%2B%2B/factorial.cpp) | [Java](/Code/Java/factorial.java)
 * Fibonocci ----> [C++](https://github.com/Algo-Phantoms/Algo-Tree/blob/main/Code/C%2B%2B/fibonacci.cpp)
 * Tower of Hanoi ----> [C++](/Code/C++/tower_of_hanoi.cpp)
+* Kth_Symbol_Grammar ----> [C++](https://github.com/Ayush12062000/Algo-Tree/blob/issue-645/Code/C%2B%2B/Kth_Symbol_Grammar.cpp)
```
Comment (krishna-NIT) [inline Algorithm/Recursion/readme.md:21]: rearrange them in alphabetical order

File: Code/C++/Kth_Symbol_Grammar.cpp [hunk 4]
```diff
@@ -59,14 +65,18 @@ int main()
...
+INPUT: 3 4
+OUTPUT: 0
+INPUT: 2 2
+OUTPUT: 1
+INPUT: 2 1
+OUTPUT: 0
+
+
 
+Time Complexity: O(n+k)
```
Comment (krishna-NIT) [inline Code/C++/Kth_Symbol_Grammar.cpp:80]: add what is n and k here ?

Comment (Ayush12062000) [reply Code/C++/Kth_Symbol_Grammar.cpp]: done

File: Algorithm/Recursion/readme.md (hunk 3 above)
Comment (Ayush12062000) [reply Algorithm/Recursion/readme.md:20]: done

**Comment by krishna-NIT** [comment]
resolve conflict

**Comment by Ayush12062000** [comment]
tried to resolve the issue.. please have a look.

never encounter such kind of conflict so I don't know what actually happened.

**Comment by krishna-NIT** [comment]
@Ayush12062000 Good Efforts,
@rish-singhal LGTM, Please Merge it!

**Pull request merged**